TSPL.docc/ReferenceManual/Statements.md

```diff
@@ -873,6 +876,47 @@ do {
 }
 ```
 
+A `do` statement can optionally specify the type of error it throws,
+which has the following form:
+
+```swift
+do throws(<#type#>) {
+    try <#expression#>
+} catch <#pattern> {
+    <#statements#>
+} catch {
+    <#statements#>
+}
+```
+
+<!-- XXX Is "do throws { }" allowed? -->
+
+If the `do` statement includes a `throws` clause,
+the `do` block can throw errors of only the specified *type*.
+The *type* must be
+a concrete type that conforms to the `Error` protocol,
+an opaque type that conforms to the `Error` protocol,
+or the boxed protocol type `any Error`.
+If the `do` statement doesn't specify the type of error it throws,
+Swift infers the error type as follows:
+
+- If every `throws` statement and `try` expression in the `do` code block
+  is nested inside of an exhaustive error-handling mechanism,
+  then Swift infers that the `do` statement is nonthrowing.
+
+- If the `do` code block contains code that throws
+  errors of only a single type
```

Comment:
A single type "other than `Never`"? Do we need to call that out here?

Reply:
Added in line 908